src/os/bluestore/BlueFS.h

```diff
@@ -341,10 +341,16 @@ class BlueFS {
 			  uint64_t jump_to = 0);
   uint64_t _estimate_log_size();
   bool _should_compact_log();
-  uint64_t _default_bluefs_alloc_size() {
-    if (cct->_conf->bluefs_alloc_size > 0)
+  uint64_t _default_bluefs_alloc_size(uint64_t bluestore_min_alloc_size) {
+    uint64_t min_allowed_size = 16384U; // same as bluestore_min_alloc_size_ssd
+    if (cct->_conf->bluefs_alloc_size > 0 &&
+	  cct->_conf->bluefs_alloc_size >= min_allowed_size)
       return cct->_conf->bluefs_alloc_size;
-    return 1024 * 1024U;
+    else if (bluestore_min_alloc_size > 0 &&
+	       bluestore_min_alloc_size >= min_allowed_size)
+      return bluestore_min_alloc_size;
+    else
+      return min_allowed_size;
```

Comment:
So for bluestore_tool one gets bluefs_alloc_size = min_allowed_size = 16K unless _conf->bluefs_alloc_size is set. Which is generally incorrect and potentially dangerous.

Reply:
I think we have two constraints on min_allowed_size here:

1) The performance impact constraint as the value goes smaller.  We could easily make it 8K but I suspect the performance impact would be pretty severe given what we are seeing with bluefs_alloc_size = 16k

2) The current constraint imposed by _get_ondisk_reserved()

I wonder if perhaps _get_ondisk_reserved() should be renamed and return whatever the minimum allowed size is instead so we don't sort of have two different constraints like this?

Reply:
@ifed01 added another commit to make the bluestore_tool use 1MB as default, that should be less harmful, right? I do not see any better way other than this, since we've agreed we do not want to persist bluefs_alloc_size.

Reply:
@neha-ojha - finally I'd rather prefer to have 0 as BlueFS ctor parameter in bluestore_tool and introduce another assert(bluefs_alloc_size != 0) in e.g. _init_alloc().
Just to avoid any senseless values in the code. And to enable BlueFS working without properly initialized alloc_size while this is possible (i.e. before the actual need for allocator).

Reply:
Sounds good to me!